pysparkling/sql/expressions/mappers.py

```diff
@@ -49,14 +51,24 @@ def __init__(self, item, field):
         self.field = field
 
     def eval(self, row, schema):
-        item_value = self.item.eval(row, schema)
+        try:
+            idx = schema.names.index(self.item.col_name)
```

Comment:
I don't believe that this approach works when there are multiple columns with the same name, for instance:

```
>>> df1 = spark.createDataFrame([("Alice", {"age":1})], ["name", "info"])
>>> df2 = spark.createDataFrame([("Alice", {"height": 1})], ["name", "info"])
>>> df1.join(df2, on=df1.name==df2.name).show()
+-----+----------+-----+-------------+
| name|      info| name|         info|
+-----+----------+-----+-------------+
|Alice|[age -> 1]|Alice|[height -> 1]|
+-----+----------+-----+-------------+

>>> df1.join(df2, on=df1.name==df2.name).select(df1.info.age).show()
+---------+
|info[age]|
+---------+
|        1|
+---------+

>>> df1.join(df2, on=df1.name==df2.name).select(df2.info.height).show()
+------------+
|info[height]|
+------------+
|           1|
+------------+

```

`column.find_position_in_schema(schema)` has the behaviour you want

Reply:
Yes the field getter is not yet complete. I think we need more unit tests there to cover all use cases. I'll try to include some today.